### PR TITLE
[Notifier] Remove superfluous parameters in *Message::fromNotification()

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 5.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Those classes are final so I think we don't need those extra arguments.

### CODE REVIEW

File: src/Symfony/Component/Notifier/Message/EmailMessage.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public function __construct(RawMessage $message, Envelope $envelope = null)
         $this->envelope = $envelope;
     }
 
-    public static function fromNotification(Notification $notification, Recipient $recipient, string $transport = null): self
```
Comment (fancyweb) [inline src/Symfony/Component/Notifier/Message/EmailMessage.php]: Alternative: for the $transport, we could set the transport directly from this method then? But the current code uses the setter everytime.

File: src/Symfony/Component/Notifier/Message/ChatMessage.php [hunk 2]
```diff
@@ -32,7 +31,7 @@ public function __construct(string $subject, MessageOptionsInterface $options =
         $this->options = $options;
     }
 
-    public static function fromNotification(Notification $notification, Recipient $recipient, string $transport = null): self
+    public static function fromNotification(Notification $notification): self
```
Comment (fancyweb) [inline src/Symfony/Component/Notifier/Message/ChatMessage.php]: Maybe we need $recipient here in the future? But being forced to pass an useless argument is worse in practice for me.

**Comment by nicolas-grekas** [comment]
There is a failure on deps=low

File: src/Symfony/Component/Notifier/Bridge/Slack/composer.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "require": {
         "php": "^7.2.5",
         "symfony/http-client": "^4.3|^5.0",
-        "symfony/notifier": "~5.0.0"
+        "symfony/notifier": "~5.0.3"
```
Comment (fancyweb) [inline src/Symfony/Component/Notifier/Bridge/Slack/composer.json]: This is wrong, I need to rebase this on master.

**Review by fabpot — COMMENTED** [review]
I did that on purpose. I'm :-1: for this change.

**Comment by fancyweb** [comment]
@fabpot In this case, shouldn't the methods at least set the passed $transport? I don't understand how keeping any unused argument can be better for usability. Especially the mandatory $recipient on ChatMessage. Could you explain your intention please? Thank you :)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @fancyweb.

**Pull request merged**